dali/python/nvidia/dali/_debug_mode.py

```diff
@@ -790,8 +790,7 @@ def run(self):
         """Run the pipeline and return the result."""
         import numpy as np
 
-        if not self._built:
-            raise RuntimeError("Pipeline must be built first.")
+        self._ensure_built()
```

Comment:
```suggestion
        self.build()
```
There's no point in having a redundant check (build is already no-op if built) and another function call.